src/Core/Domain/Product/QueryResult/ProductBaseData.php

```diff
@@ -0,0 +1,70 @@
+<?php
+/**
+ * 2007-2020 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2020 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Domain\Product\QueryResult;
+
+use PrestaShop\PrestaShop\Core\Domain\Product\ValueObject\ProductType;
+
+class ProductBaseData
+{
+    /**
+     * @var string[]
+     */
+    private $localizedNames;
+
+    /**
+     * @var ProductType
+     */
+    private $productType;
+
+    /**
+     * @param string[] $localizedNames
+     * @param int $productType
+     */
+    public function __construct(array $localizedNames, int $productType)
+    {
+        $this->localizedNames = $localizedNames;
+        $this->productType = new ProductType($productType);
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getLocalizedNames(): array
+    {
+        return $this->localizedNames;
+    }
+
+    /**
+     * @return ProductType
+     */
+    public function getProductType(): ProductType
```

Comment:
Is it necessary ? 🤔
I'm wondering ... since it will always be an integer, do we need a VO ?

Reply:
idk where to keep those integers constants then and make sure its valid?